app/controllers/api/v1/customers_controller.rb

```diff
@@ -61,6 +72,11 @@ def authorize_action
       def search_customers
         customers = visible_customers.includes(:bill_address, :ship_address)
         customers = customers.where(enterprise_id: params[:enterprise_id]) if params[:enterprise_id]
+
+        if @extra_customer_fields.include?(:balance)
+          customers = CustomersWithBalance.new(customers: customers).query
+        end
```

Comment:
Repeating a comment from my first review: This is generating a complete new query based on customer ids. It resets the `includes(:bill_address, :ship_address)` and could have a negative impact on performance. Wouldn't it be better to take the existing query and add the customer balance part on top?

Reply:
Missed that one. Very good thinking, fixed, much cleaner now